olp-cpp-sdk-core/tests/thread/TaskContinuationTest.cpp

```diff
@@ -174,31 +143,23 @@ TEST_F(TaskContinuationFixture, CallExecute) {
               return olp::client::CancellationToken();
             });
           })
-          .Finally([&promise,
-                    &result](olp::client::ApiResponse<olp::client::HttpResponse,
-                                                      olp::client::ApiError>
-                                 response) {
-            result = response.GetResult();
-            promise.set_value();
-          });
+          .Finally(
+              [&promise](ResponseType<olp::client::HttpResponse> response) {
+                promise.set_value(response.MoveResult());
+              });
 
   cont.Run();
 
-  EXPECT_EQ(future.wait_for(std::chrono::milliseconds(1000)),
-            std::future_status::ready);
-
-  future.get();
+  auto result = future.get();
 
   EXPECT_TRUE(result.IsSuccessful());
+  EXPECT_EQ(result.GetResult().GetStatus(), 200);
 }
 
 TEST_F(TaskContinuationFixture, CallCancel) {
-  std::promise<void> promise;
+  std::promise<ResponseType<olp::client::HttpResponse>> promise;
```

Comment:
Please use the approach described above.

Reply:
done